krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -182,28 +176,23 @@ func (r *itfce) SetSpec(spec *nephioreqv1alpha1.InterfaceSpec) error {
 		if err := r.SetNetworkInstanceName(string(spec.NetworkInstance.Name)); err != nil {
 			return err
 		}
-	} else {
-		return fmt.Errorf("networkInstance is required")
 	}
 	return nil
 }
 
 // DeleteAttachmentType deletes the attachmentType from the spec
 func (r *itfce) DeleteAttachmentType() error {
-	return r.deleteNestedField(attachmentType...)
+	return r.deleteNestedField(cniType...)
```

Comment:
Looks like a typo here. "attachmentType" needs to be passed in.

Reply:
fixed + added a test to cover this